src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -491,12 +491,19 @@ public override Task<bool> HandleRequestAsync()
                 return HandleRequestResult.Fail("No message.");
             }
 
+            AuthenticationProperties properties = null;
             try
             {
-                AuthenticationProperties properties = null;
                 if (!string.IsNullOrEmpty(authorizationResponse.State))
                 {
                     properties = Options.StateDataFormat.Unprotect(authorizationResponse.State);
+
```

Comment:
Nit: private helper method for this Unprotect/decode clear state block that's repeated?